src/main/kotlin/it/pagopa/ecommerce/helpdesk/utils/ResponseBuilderV2.kt

```diff
@@ -0,0 +1,146 @@
+        else -> getTransactionFee(baseTransaction)
+    }
+
+private fun getTransactionAuthRequestedData(
+    baseTransaction: BaseTransaction
+): TransactionAuthorizationRequestData? =
+    when (baseTransaction) {
+        is BaseTransactionExpired ->
+            getTransactionAuthRequestedData(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithClosureError ->
+            getTransactionAuthRequestedData(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithRequestedAuthorization ->
+            baseTransaction.transactionAuthorizationRequestData
+        else -> null
+    }
+
+private fun getTransactionAuthCompletedData(
+    baseTransaction: BaseTransaction
+): TransactionAuthorizationCompletedData? =
+    when (baseTransaction) {
+        is BaseTransactionExpired ->
+            getTransactionAuthCompletedData(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithClosureError ->
+            getTransactionAuthCompletedData(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithRefundRequested ->
+            getTransactionAuthCompletedData(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithCompletedAuthorization ->
+            baseTransaction.transactionAuthorizationCompletedData
+        else -> null
+    }
+
+private fun getTransactionUserReceiptData(
+    baseTransaction: BaseTransaction
+): TransactionUserReceiptData? =
+    when (baseTransaction) {
+        is BaseTransactionExpired ->
+            getTransactionUserReceiptData(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithRefundRequested ->
+            getTransactionUserReceiptData(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithRequestedUserReceipt -> baseTransaction.transactionUserReceiptData
+        else -> null
+    }
+
+private fun getTransactionDetailsStatus(baseTransaction: BaseTransaction): String =
+    when (getAuthorizationOutcomeV2(baseTransaction)) {
+        AuthorizationResultDto.OK -> "Confermato"
+        AuthorizationResultDto.KO -> "Rifiutato"
+        else -> "Cancellato"
+    }
+
+fun getAuthorizationOutcomeV2(baseTransaction: BaseTransaction): AuthorizationResultDto? =
+    when (baseTransaction) {
+        is BaseTransactionExpired ->
+            getAuthorizationOutcomeV2(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithRefundRequested ->
+            getAuthorizationOutcomeV2(baseTransaction.transactionAtPreviousState)
+        is TransactionWithClosureError ->
+            getAuthorizationOutcomeV2(baseTransaction.transactionAtPreviousState)
+        is BaseTransactionWithCompletedAuthorization ->
+            // TODO check v1/v2 mapping for authorization result
+            if (baseTransaction.transactionAuthorizationCompletedData.authorizationCode != null)
+                AuthorizationResultDto.OK
+            else AuthorizationResultDto.KO
```

Comment:
```suggestion
         is BaseTransactionWithCompletedAuthorization -> {
            val gatewayAuthData =
                baseTransaction.transactionAuthorizationCompletedData.transactionGatewayAuthorizationData
            when (gatewayAuthData) {
                is PgsTransactionGatewayAuthorizationData -> gatewayAuthData.authorizationResultDto
                is NpgTransactionGatewayAuthorizationData -> if (gatewayAuthData.operationResult == OperationResultDto.EXECUTED) {
                    AuthorizationResultDto.OK
                } else {
                    AuthorizationResultDto.KO
                }
            }
        }
```
Maybe here authorization result enum value can be extracted by something like that (discriminating between PGS and NPG used gateway)